template/READ_BEFORE_COMMIT.md.tpl

```diff
@@ -0,0 +1,9 @@
+# Instructions
+
+This document lays out the next steps for AWS service teams upon bootstrapping the ACK {{ .ServicePackageName }}-controller repository successfully.
+
+1. To get started, look inside the `generator.yaml` file, and remove the resource(s) from the ignore list and execute `make build-controller` from the ACK code-generator.
+2. Update service full_name and short_name inside `metadata.yaml` file.
+2. Add any custom code inside the `templates` directory.
+3. Refer to testing documentation to add e2e test inside the `test` directory.
+4. Add your team members' GitHub aliases to your team controller alias in `OWNERS_ALIASES` file.
```

Comment:
I think these instructions need to be a lot more detailed. We need to tell them not only what, but why and how. This will more easily let us consistently share information for new repository owners and reduce the amount we have to repeat ourselves.

```suggestion
1. To get started, edit `generator.yaml` file and comment out (with a `#`) each of the resource(s) from the ignore list, generating your controller after every edit.

  This step will enable each resource within the service to be generated into an ACK custom resource definition. As you generate each of them, the code-generator may require additional configuration (in the `generator.yaml` file) in order for you to continue.
  
2. Update fields under `service` inside `metadata.yaml`

  This file provides display information about the controller and the service it supports. This file is intended to be manually updated to ensure it matches the proper casing and terminology for the service. For `full_name`, provide the full display name of the service (eg. `Amazon Simple Storage Service`). For `short_name`, provide the abbreviation or shortened service name (eg. `S3` or `SageMaker` - do not include `AWS` or `Amazon`). For `link`, provide a link to the homepage for the service. For `documentation`, provide a link to the homepage for the service documentation/user guide.
  
3. Add your team members' GitHub aliases to the `OWNERS_ALIASES` file

  This file determines who has permissions to review and merge code within the repository.

4. Add any custom code to the `templates` directory

  As you work on the controller, you may need to add custom hooks to support functionality specific to your service. Hook code for any resource should be placed in a file with the path `templates/hooks/<resource name>/<hook name>.go.tpl`
  
5. Refer to the open source documentation on testing for creating and running tests, which should be added in the `test/e2e` directory.

  All PRs containing changes to CRDs should be accompanied with appropriate tests for any new functionality or custom code. Documentation for these tests can be found in the "Contributor Docs" section of the ACK website.
```

Reply:
Yes, thank you for pointing this out. The instructions have been updated with the suggested details and information.